django/db/backends/sqlite3/introspection.py

```diff
@@ -192,9 +192,9 @@ def get_primary_key_column(self, cursor, table_name):
         fields_sql = create_sql[create_sql.index('(') + 1:create_sql.rindex(')')]
         for field_desc in fields_sql.split(','):
             field_desc = field_desc.strip()
-            m = re.search('"(.*)".*PRIMARY KEY( AUTOINCREMENT)?', field_desc)
+            m = re.match(r"\"?([\w-]+)\"?.*PRIMARY KEY(?: AUTOINCREMENT)?.*", field_desc)
```

Comment:
I don't see the reason for switching to double quotes -- that just seems to add the requirement to escape the double quotes in the regex. Also, why the `?:` and `.*` changes? I don't see test failures with those changes reverted. Also no failures if the `-` after `\w` is removed. Please try to keep the regex as simple as possible and minimize the number of changes, or at least explain all the changes.

Reply:
I think I'm being overly greedy in my regex as you've pointed out. A simple `([\w-]+).*PRIMARY KEY` would suffice here.